### DISCUSSION TITLE
Why is user not a domain object?

### DISCUSSION THREAD

**Discussion by TCrosmanDSI** [jasontaylordev/CleanArchitecture#489] [Q&A]
Excuse my ignorance as I'm sure you have a good reason.  I am a little confused as to why ApplicationUser is not a domain object.  Perhaps it's because you are only using the user object for record keeping?

But say I wanted to assign a ToDo to a user in this design, I don't currently see a way to foreign key that relationship.

Thoughts?

**Comment by AlexAlexGoTO** [comment]
Just investigated this architecture and have the same question, how to add any one-to-one relationship to user User>ToDo... ?
In my case user should has list of tasks.

P.S. I just added one more project to my solution with all entities. But I would like to hear better solution :)

**Comment by AlexAlexGoTO** [comment]
> Excuse my ignorance as I'm sure you have a good reason. I am a little confused as to why ApplicationUser is not a domain object. Perhaps it's because you are only using the user object for record keeping?
> 
> But say I wanted to assign a ToDo to a user in this design, I don't currently see a way to foreign key that relationship.
> 
> Thoughts?

Okay mate, I got it.
![image](https://user-images.githubusercontent.com/18060095/142290645-a2d96f40-0c21-40e1-a08b-f203f01afa49.png)
It works in a bit different way Infrastructure has reference to Application and Application has reference to Domain. So currently we don't need to have direct reference to Domain to use ToDoItems inside ApplicationUser. It's like has access to everything that Application layer has...

**Reply by TCrosmanDSI** [reply]
Neat I didn't know it would allow that.  I'll test it out myself when I get home.

Did you do a database migration and run update-database with out any issues?

Often times the identity server is on a separate sql server, so if that was the case I don't think this would be possible.

**Comment by TCrosmanDSI** [comment]
You won't be able to reference the infrastructure project from the domain project, so your relationships would only be exposed through the user object.

So I think so far the only option is to add a user object to domain that is different from the identity user.

Any other thoughts on this?

**Reply by artworkad** [reply]
It's absolutely fine to put a user entity into the core domain. However I would give the entity a name, that expresses its meaning. E.g. Author (still pretty bad in the todo context :D). However if the user is just used for authentication and authorization, I would keep it out of the domain.
